src/wp-includes/class-wp-script-modules.php

```
@@ -23,12 +23,12 @@ class WP_Script_Modules {
 	private $registered = array();
 
 	/**
-	 * Holds the script module identifiers that were enqueued before registered.
+	 * An array of IDs for queued script modules.
```

Why is there a need to switch from keying by script ID to an array of script ID values? 

It's adding a bunch of work to avoid duplicates that I think could be more easily covered by using a method to add the new asset in the wp_block class. 

As developer interfaces go, I also think that an `::add_to_queue` method would be more helpful that requiring third party devs do an `array_unique( array_merge() )` each time they modify the queue.

Even if we advice that manipulating the queue in this manner is discouraged, it will be done.

---

This brings `WP_Script_Modules` in alignment with `WP_Scripts` and `WP_Styles` which both expose a `$queue` member variable. It allows us to easily obtain the list of IDs enqueued, and then merge them.

For `WP_Scripts` and `WP_Styles` it is primarily for reading, but it is also used for writing here (now in 6.9):

https://github.com/WordPress/wordpress-develop/blob/b3c3c8e518d462f821e29edb7a95accd9f2b474b/src/wp-includes/media.php#L2113-L2114

In reality, the `array_unique()` is not needed because `WP_Scripts`, `WP_Styles`, and `WP_Script_Modules` all obtain the unique items when processing anyway (or else keep track via `done`).

We could introduce a magic setter function for the `queue` member (which we could then make private). This would have the benefit of not only ensuring uniqueness, but we could also throw an error if anything other than an array of strings is attempted to be set.

---

Thanks, let's go with consistency.